Source/engine/render/text_render.cpp

```diff
@@ -275,6 +275,11 @@ void WordWrapGameString(char *text, size_t width, GameFontTables size, int spaci
 	}
 }
 
+int DrawString(const CelOutputBuffer &out, const char *text, const Point &position, uint16_t flags, int spacing, int lineHeight, bool drawTextCursor)

```

Comment:
Now I feel like there is a ton of places that should start using this :D

Reply:
So it turns out default initialisers and implicitly defined constructors combine to make `{1, 2}` a valid initialiser for both `Point` and `SDL_Rect`. In the case of `SDL_Rect` the values populate x and y, leaving w and h to get default-initialised (with the value 0). This means that with `DrawString(..., Point, ...)` and `DrawString(..., SDL_Rect, ...)` both available, a call such as `DrawString(..., {2, 4}, ...)` is ambiguous as it could be interpreted as `DrawString(..., SDL_Rect{2, 4/*, 0, 0*/}, ...)` or `DrawString(..., Point{2, 4}, ...)`. Using a previously defined variable of type `SDL_Rect` or `Point` (or a type with an implicit conversion to only one of those types) is unambiguous.

If only the `SDL_Rect` version of the function was available and there was an implicit conversion from `Point` to `SDL_Rect` then the behaviour would be transparent to callers, but this can be confusing. With the explicit conversion function from the previous commit callers need to do `static__cast<SDL_Rect>(Point)` which at least cuts down on the confusion a bit, but looks ugly.

Looking at the callers, DrawString is called in three "modes"; with a position only (width/height of 0), with a position and a width to force specific wrapping (height 0), or with a full bounding rect. Thankfully, only the second case is ever called with hardcoded x/y values, every case using purely a position is better represented with a `Point` variable declared earlier in the calling function.

With that in mind I'll update this pull request after refactoring the "position only" call sites to use Points, I think this will be reasonable.

Reply:
I've updated this request with every instance where DrawString was called with only x,y params changed to use the Point version of the function.